src/Adapter/Order/QueryHandler/GetOrderProductsForViewingHandler.php

```diff
@@ -169,7 +169,8 @@ public function handle(GetOrderProductsForViewing $query): OrderProductsForViewi
 
         unset($product);
 
-        ksort($products);
+        // reorder products by order_detail_id DESC
+        krsort($products);
```

Comment:
I think this should be related to a parameter to indicate the expected order, optional parameter that you can set in the OrderController

This would prevent modifying the test that need to reorder the result to get the expected results I wouldn't go as far as classifying this as a BC break (to change the order) but at least it should be choosable (and the default behaviour should be the original one)